cluster/cluster.go

```diff
@@ -140,9 +145,10 @@ LOOP:
 				continue LOOP
 			}
 		}
-		// if no nodes have been selected yet then grab a
-		// random node from the set of available nodes
-		selected := candidates.nodes[rand.Intn(len(candidates.nodes))]
+		// if no nodes have been selected yet then grab a node from
+		// the set of available nodes in such a way that nodes are
+		// weighted fairly across MembersForQuery calls
+		selected := candidates.nodes[count%len(candidates.nodes)]
```

Comment:
is reading the count like this atomic?

Reply:
ah, sorry, got it now. the `count` is the return value of the above atomic operation